pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -97,6 +98,15 @@ type ManagerImpl struct {
 
 	// pendingAdmissionPod contain the pod during the admission phase
 	pendingAdmissionPod *v1.Pod
+
+	// containerMap provides a mapping from (pod, container) -> containerID
+	// for all containers a pod. Used to detect pods running across a restart
+	containerMap containermap.ContainerMap
+
+	// containerRunningSet identifies which container among those present in `containerMap`
+	// was reported running by the container runtime when `containerMap` was computed.
+	// Used to detect pods running across a restart
+	containerRunningSet sets.String
```

Comment:
I'm normally not a huge fan of adding a one-off, special purpose list like this, but I don't see a good alternative other than to maybe turn this into a list of the full `runtimeapi.Container` objects. That way we can pull other information from these objects in the future if necessary.

Reply:
Same considerations as per https://github.com/kubernetes/kubernetes/pull/118635#discussion_r1261023770 apply. TL;DR: this code reorganization is likely worthy, but deserves its own separate effort.